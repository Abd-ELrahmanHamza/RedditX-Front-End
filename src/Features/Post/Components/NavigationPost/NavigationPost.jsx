// import Components
import React, { useState } from "react";
import Modal from "react-bootstrap/Modal";
import {
  CloseButton,
  CloseButtonContainer,
  Down,
  FlairTitle,
  Gallery,
  Header,
  Rate,
  Title,
  Up,
} from "./NavigationPost.styled";
import { TbArrowBigTop, TbArrowBigDown } from "react-icons/tb";
import { RiGalleryLine } from "react-icons/ri";
import { Link, useNavigate } from "react-router-dom";
import { IoMdClose } from "react-icons/io";

/**
 * Component that displays a post page displayed on modal window
 * @param setHandleShowModal - handle show modal window which is true by clicking on post and false by clicking on close button
 * @return {React.Component}
 */
const NavigationPost = ({ setHandleShowModal }) => {
  const navigate = useNavigate();
  const [count, setCount] = useState(0);
  const addCount = () => {
    let countTemp = count + 1;
    setCount(countTemp);
  };
  const deleteCount = () => {
    let countTemp = count - 1;
    if (countTemp >= 0) setCount(countTemp);
    else setCount(0);
  };

  return (
    <Modal.Header>
      <Header>
        <div className={"content"}>
          <Rate>
            <Up onClick={addCount}>
              <TbArrowBigTop size={26} />
            </Up>
            <div className={"score"}>{count}</div>
            <Down onClick={deleteCount}>
              <TbArrowBigDown size={26} />
            </Down>
          </Rate>
          <Gallery>
            <RiGalleryLine />
          </Gallery>
          <Title>
            <h6>الواحد و هو بيتفرج مش عارف يفرح ولا يزعل</h6>
          </Title>
          <FlairTitle>
            <div>
              <Link
                href={
                  "https://www.reddit.com/r/Egypt/search?q=flair_name%3A%22Sports%20%D8%B1%D9%8A%D8%A7%D8%B6%D8%A9%20%22&restrict_sr=1"
                }
              >
                <div className={"text-flair"}>
                  <span>Sports رياضة </span>
                </div>
              </Link>
            </div>
          </FlairTitle>
        </div>
        {/*<CloseButtonContainer onClick={()=>setHandleShowModal(false)}>*/}
        <CloseButtonContainer onClick={() => navigate(-1)}>
          <CloseButton>
            <i>
              <IoMdClose />
            </i>
            <span>Close</span>
          </CloseButton>
        </CloseButtonContainer>
      </Header>
    </Modal.Header>
  );
};

export default NavigationPost;
