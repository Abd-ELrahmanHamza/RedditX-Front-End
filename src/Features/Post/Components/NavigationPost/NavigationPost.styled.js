import styled from "styled-components";

// styles of heading
export const Header = styled.div`
  align-items: center;
  box-sizing: border-box;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  margin: auto;
  max-width: 1128px;
  padding: 0 12px 0 32px;
  width: 100%;

  @media (min-width: 960px) {
    padding: 0 32px;
  }
  @media (max-width: 550px){
    padding: 0;
  }

  // content styles
  .content {
    align-items: center;
    display: -ms-flexbox;
    display: flex;
    flex: 1;
    max-width: calc(100% - 324px);
    width: 100%;
    
  }
`
// rate container
export const Rate = styled.div`
  align-items: center;;
  display: flex;
  flex-direction: row;
  margin: 0;
  padding: 0 2px;


  // styles of up button and before
  .but-up {
    height: 24px;
    width: 24px;
    background: transparent;
    border: none;
    color: inherit;
    cursor: pointer;
    padding: initial;
  }

  :before {
    border-right: 1px solid #343536;
    content: "";
    display: inline-block;
    height: 16px;
    vertical-align: text-bottom;
    width: 0;
    margin-right: 8px;

  }

  // styles of down button and after
  .but-down {
    background: transparent;
    height: 24px;
    width: 24px;
    border: none;
    color: inherit;
    cursor: pointer;
    padding: initial;
  }

  :after {
    border-right: 1px solid #343536;
    content: "";
    display: inline-block;
    height: 16px;
    vertical-align: text-bottom;
    width: 0;
    margin: 0 8px;
  }


  // number of likes
  .score {
    line-height: 15px;
    margin: 0 1px;
    text-align: center;
    width: 32px;
    color: rgb(215, 218, 220);
    font-size: 12px;
    font-weight: 700;
    pointer-events: none;
    word-break: normal;
    padding-top: 8px;
  }
`
export const Arrow = styled.span
  `
      color: white;
      display: inline-block;
      overflow: hidden;
      height: 24px;
      width: 24px;
      font-size: 16px;
      line-height: 24px;
      position: relative;
    `

// styles of Gallery img element
export const Gallery = styled.span
  `
      color: #d7dadc;
      font-size: 16px;
      margin-right: 8px;
      @media (max-width: 550px){
        display:none;
      }
    `
export const Title = styled.div
  `
      overflow: hidden;
      text-overflow: ellipsis;
      white-space: nowrap;
      display: flex;
      min-width: 0;
      font-size: 14px;
      font-weight: 300;
      line-height: 18px;
      margin-right: 3px;

      h6 {
        display: inline-block;
        color: white;
        font-weight: inherit;
        margin-bottom: 0.2rem;
      }
      @media(max-width: 1070px){
        display:none;
      }
    `
// styles of category title
export const FlairTitle = styled.div
  `
      display: inline-flex;
      margin: 0;
      padding: 0;
      vertical-align: text-top;

      :first-child {
        display: inline-block;
      }

      a {
        text-decoration: inherit;
        text-decoration: inherit;
      }

      .text-flair {
        font-size: 12px;
        font-weight: 400;
        line-height: 16px;
        border-radius: 2px;
        display: inline-block;
        margin-right: 5px;
        overflow: hidden;
        text-overflow: ellipsis;
        vertical-align: text-bottom;
        white-space: pre;
        word-break: normal;
        margin-left: 0;
        padding: 0 4px;
        background-color: rgb(0, 166, 165);
        color: rgb(255, 255, 255);
      }
      @media (max-width:550px){
        display: none;
      }
    `

export const CloseButtonContainer = styled.div
  `
      display: flex;
      -ms-flex-pack: end;
      justify-content: flex-end;
      margin-left: 12px;
      width: 312px;
      font-size: 12px;
      font-weight: 700;
      line-height: 16px;
    `
export const CloseButton = styled.button
  `
      background: transparent;
      color: #d7dadc;
      position: relative;
      border: 1px solid transparent;
      font-family: Noto Sans, Arial, sans-serif;
      font-size: 12px;
      font-weight: 700;
      letter-spacing: unset;
      line-height: 16px;
      text-transform: unset;
      min-height: 24px;
      min-width: 24px;
      padding: 4px 8px;
      box-sizing: border-box;
      display: flex;
      justify-content: center;
      text-align: center;
      width: auto;
      align-items: baseline;

      i {
        display: inline-block;
        height: 16px;
        width: 16px;
        font-size: 20px;
        line-height: 16px;
        padding: 0 4px 0 0;
        margin-right: 4px;
      }
    `