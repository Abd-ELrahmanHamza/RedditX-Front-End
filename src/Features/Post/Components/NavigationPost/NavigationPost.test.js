import { shallow } from "enzyme";
import React from "react";

// Import components
import NavigationPost from "./NavigationPost";

const mockedUsedNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useNavigate: () => mockedUsedNavigate,
}));

describe("Navigation Post", () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<NavigationPost />);
    });

    it("should render without crashing", () => {
        expect(wrapper).toMatchSnapshot();
    });
});
