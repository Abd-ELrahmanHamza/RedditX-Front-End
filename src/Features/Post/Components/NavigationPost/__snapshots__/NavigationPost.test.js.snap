// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Post should render without crashing 1`] = `
<ModalHeader
  closeButton={false}
  closeLabel="Close"
>
  <styled.div>
    <div
      className="content"
    >
      <styled.div>
        <styled.button
          onClick={[Function]}
        >
          <TbArrowBigTop
            size={26}
          />
        </styled.button>
        <div
          className="score"
        >
          0
        </div>
        <styled.button
          onClick={[Function]}
        >
          <TbArrowBigDown
            size={26}
          />
        </styled.button>
      </styled.div>
      <styled.span>
        <RiGalleryLine />
      </styled.span>
      <styled.div>
        <h6>
          Post Header
        </h6>
      </styled.div>
      <styled.div>
        <div>
          <Link
            href="https://www.reddit.com/r/Egypt/search?q=flair_name%3A%22Sports%20%D8%B1%D9%8A%D8%A7%D8%B6%D8%A9%20%22&restrict_sr=1"
          >
            <div
              className="text-flair"
            >
              <span>
                Sports رياضة 
              </span>
            </div>
          </Link>
        </div>
      </styled.div>
    </div>
    <styled.div
      onClick={[Function]}
    >
      <styled.button>
        <i>
          <IoMdClose />
        </i>
        <span>
          Close
        </span>
      </styled.button>
    </styled.div>
  </styled.div>
</ModalHeader>
`;
